src/app/tree-grid/tree-grid-row-reorder/tree-grid-row-reorder.component.ts

```diff
@@ -0,0 +1,61 @@
+import { Component, ViewChild } from "@angular/core";
+import { IgxDropEventArgs, IgxTreeGridComponent, IgxTreeGridRowComponent, Point } from "igniteui-angular";
+import { FULL_EMPLOYEE_DATA } from "../data/employees";
+
+@Component({
+    selector: "tree-grid-row-reorder",
+    styleUrls: ["tree-grid-row-reorder.component.scss"],
+    templateUrl: "tree-grid-row-reorder.component.html"
+})
+export class TreeGridRowReorderComponent {
+    @ViewChild(IgxTreeGridComponent, { read: IgxTreeGridComponent })
+    public treeGrid: IgxTreeGridComponent;
+
+    public localData = [];
+    constructor() {
+        this.localData = FULL_EMPLOYEE_DATA();
+    }
+
+    public rowDragStart(args: any): void {
+        const targetRow: IgxTreeGridRowComponent = args.dragData;
+        if (targetRow.expanded) {
+            const grid = targetRow.grid;
+            grid.expansionStates = grid.expansionStates.set(targetRow.rowID, false);
+        }
+    }
+
+    public dropInGrid(args: IgxDropEventArgs): void {
+        args.cancel = true;
+        const draggedRow: IgxTreeGridRowComponent = args.dragData;
+        const event = args.originalEvent;
+        const cursorPosition: Point = { x: event.pageX, y: event.pageY };
+        this.moveRow(draggedRow, cursorPosition);
```

Comment:
This handling of `moveRow` leads to some issues in the case of dragging a child row inside of another parent row.
You should handle row moving depending on the dropped row's `parentID`